File: code-coaches/medium/cmyk-to-rgb/CmykToRgb.swift
import Foundation

func toRGB(cyan: Float, magenta: Float, yellow: Float, key: Float) -> String {
    let red = Int(round(255 * (1 - cyan) * (1 - key)))
    let green = Int(round(255 * (1 - magenta) * (1 - key)))
    let blue = Int(round(255 * (1 - yellow) * (1 - key)))

    return "\(red),\(green),\(blue)"
}

func main() {
    if let cyan = readLine().flatMap(Float.init),
       let magenta = readLine().flatMap(Float.init),
       let yellow = readLine().flatMap(Float.init),
       let key = readLine().flatMap(Float.init) {
        print(toRGB(cyan: cyan, magenta: magenta, yellow: yellow, key: key))
    }
}

main()

File: code-coaches/medium/cmyk-to-rgb/cmyk_to_rgb.cpp
#include <iostream>
#include <cmath>
#include <string>

std::string to_rgb(float cyan, float magenta, float yellow, float key)
{
    int red = round(255 * (1 - cyan) * (1 - key));
    int green = round(255 * (1 - magenta) * (1 - key));
    int blue = round(255 * (1 - yellow) * (1 - key));
    return std::to_string(red) + "," + std::to_string(green) + "," + std::to_string(blue);
}

int main()
{
    float cyan, magenta, yellow, key;
    std::cin >> cyan >> magenta >> yellow >> key;

    std::cout << to_rgb(cyan, magenta, yellow, key);

    return 0;
}
